src/main.c

```diff
@@ -242,12 +248,15 @@ static gboolean install_start(int argc, char **argv)
 			goto out_loop;
 		}
 		g_debug("Trying to contact rauc service");
-		if (!r_installer_call_install_sync(installer, args->name, NULL,
+		if (!r_installer_call_install_bundle_sync(installer, args->name, arg_args, NULL,
 				&error)) {
 			g_printerr("Failed %s\n", error->message);
 			g_error_free(error);
+			g_variant_unref(arg_args);
 			goto out_loop;
 		}
+
+		g_variant_unref(arg_args);
```

Comment:
I'm pretty sure `arg_args` leaks in the error case. Maybe use a cleanup handler?

Reply:
Ouh, I've missed the first error handling block, only added it to the second one.

Reply:
I've refactored this.